python/graphframes/tests.py

```diff
@@ -81,39 +88,42 @@ def stopSparkContext(cls):
 
     @classmethod
     def spark_at_least_of_version(cls, version_str):
-        assert hasattr(cls, 'spark_version')
+        assert hasattr(cls, "spark_version")
         required_version = cls.parse_spark_version(version_str)
         spark_version = cls.spark_version
-        for _name in ['major', 'minor', 'maintenance']:
+        for _name in ["major", "minor", "maintenance"]:
             sc_ver = spark_version[_name]
             req_ver = required_version[_name]
             if sc_ver != req_ver:
                 return sc_ver > req_ver
         # All major.minor.maintenance equal
         return True
 
-def setUpModule():
-    GraphFrameTestUtils.createSparkContext()
 
-def tearDownModule():
+@pytest.fixture(scope="module", autouse=True)
+def spark_context():
+    GraphFrameTestUtils.createSparkContext()
+    yield
     GraphFrameTestUtils.stopSparkContext()
 
 
-class GraphFrameTestCase(unittest.TestCase):
-
-    @classmethod
-    def setUpClass(cls):
-        # Small tests run much faster with spark.sql.shuffle.partitions = 4
-        cls.spark = SparkSession(GraphFrameTestUtils.sc).builder.config('spark.sql.shuffle.partitions', 4).getOrCreate()
+@pytest.fixture(scope="class")
+def spark_session():
+    # Create a SparkSession with a smaller number of shuffle partitions.
+    spark = (
+        SparkSession(GraphFrameTestUtils.sc)
```

Comment:
We can just explicitly pass all the same confs (app name, etc.) to the `SparkSession.builder`

Reply:
Oh, it was using the freaking `SparkContext` - roger!